### PR TITLE
[SPARK-37316][SQL] Add code-gen for existence sort merge join

### PR DESCRIPTION
**Pull request by c21**
### What changes were proposed in this pull request?

This PR is to add code-gen for Existence sort merge join. Followed the same algorithm used in iterator mode from `SortMergeJoinExec.scala` and `HashJoin.scala`. Check every left side row if there's an existing match from right side. Output every left side row and boolean flag to indicate whether it has a match or not.

In addition, to help review as this PR triggers several TPCDS plan files change. The below file is having the real code change:

* `SortMergeJoinExec.scala`

### Why are the changes needed?

To improve performance for Existence sort merge join, and conclude all join types code-gen support for sort merge join.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Existing unit test in `ExistenceJoinSuite.scala`

### CODE REVIEW

File: sql/core/src/main/scala/org/apache/spark/sql/execution/joins/SortMergeJoinExec.scala [hunk 1]
```diff
@@ -675,6 +677,9 @@ case class SortMergeJoinExec(
         bufferedVars ++ streamedVars
       case LeftSemi | LeftAnti =>
         streamedVars
+      case ExistenceJoin(_) =>
+        streamedVars ++ Seq(ExprCode.forNonNullValue(
```
Comment (c21) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/joins/SortMergeJoinExec.scala:681]: Just FYI - this is using same logic as hash join - https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/execution/joins/HashJoin.scala#L667-L668 .

**Comment by c21** [comment]
cc @cloud-fan could you help take a look when you have time? Thanks.

**Review by cloud-fan — APPROVED** [review]

**Comment by cloud-fan** [comment]
thanks, merging to master!

**Comment by c21** [comment]
Thank you @cloud-fan for review!

**Pull request closed without merging**